### ISSUE TITLE
`FileNotFoundError` after update to 5.1

### ISSUE THREAD

**Issue by sergiud** [gcovr/gcovr#587]
**Describe the bug**
After updating from gcovr 5.0 to 5.1, collecting coverage using same setup causes the following error

```console
$ cd build/
$ gcovr -r .. -s --html -o ${PROJECT_DIR}/coverage/report.html --xml-pretty --xml ${PROJECT_DIR}/cobertura.xml
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/gcovr/workers.py", line 77, in worker
    work(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 317, in process_datafile
    done = run_gcov_and_process_files(
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 490, in run_gcov_and_process_files
    process_gcov_data(fname, covdata, abs_filename, options)
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 93, in process_gcov_data
    with io.open(
FileNotFoundError: [Errno 2] No such file or directory: '/builds/sergiu/tracker/stl_pair.h##47d0af436808dced1f869194db0c00bc.gcov'
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/gcovr/workers.py", line 77, in worker
    work(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 317, in process_datafile
    done = run_gcov_and_process_files(
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 490, in run_gcov_and_process_files
    process_gcov_data(fname, covdata, abs_filename, options)
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 93, in process_gcov_data
    with io.open(
FileNotFoundError: [Errno 2] No such file or directory: '/builds/sergiu/tracker/stl_pair.h##47d0af436808dced1f869194db0c00bc.gcov'
(ERROR) Uncaught EXCEPTION
Traceback (most recent call last):
  File "/usr/sbin/gcovr", line 33, in <module>
    sys.exit(load_entry_point('gcovr==5.1', 'console_scripts', 'gcovr')())
  File "/usr/lib/python3.10/site-packages/gcovr/__main__.py", line 308, in main
    collect_coverage_from_gcov(covdata, options)
  File "/usr/lib/python3.10/site-packages/gcovr/__main__.py", line 360, in collect_coverage_from_gcov
    with Workers(
  File "/usr/lib/python3.10/site-packages/gcovr/workers.py", line 170, in __exit__
    self.wait()
  File "/usr/lib/python3.10/site-packages/gcovr/workers.py", line 161, in wait
    raise self.exceptions[0][1]
  File "/usr/lib/python3.10/site-packages/gcovr/__main__.py", line 367, in collect_coverage_from_gcov
    contexts = pool.wait()
  File "/usr/lib/python3.10/site-packages/gcovr/workers.py", line 161, in wait
    raise self.exceptions[0][1]
  File "/usr/lib/python3.10/site-packages/gcovr/workers.py", line 77, in worker
    work(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 317, in process_datafile
    done = run_gcov_and_process_files(
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 490, in run_gcov_and_process_files
    process_gcov_data(fname, covdata, abs_filename, options)
  File "/usr/lib/python3.10/site-packages/gcovr/gcov.py", line 93, in process_gcov_data
    with io.open(
FileNotFoundError: [Errno 2] No such file or directory: '/builds/sergiu/tracker/stl_pair.h##47d0af436808dced1f869194db0c00bc.gcov'
```

**To Reproduce**
I would love to provide a reproducer. However, since the project is large and the error does not provide details, I don't even know where to start looking.

**Expected behavior**
No error.

**Desktop (please complete the following information):**
 - OS: ArchLinux
 - GCC version 11.2
 - GCOVR version 5.1
 - Project directory layout:
   - `include/`
   - `src/`
   - `build/`

**Comment by Spacetown** [comment]
That's strange because gcov wrote this path as generated file to stdout.
@latk I'd like to add a debug message with the whole stdout and stderr of the gcov command. What do you think about?

**Comment by latk** [comment]
@Spacetown yes, if I learned anything from this release it's that we need better error messages :)

In principle, our `run_gcov_and_process_files()` function needs to move most of its contents into a try…except block so that we can provide context if something happens (working directories, gcda file name, gcov file name, gcov stderr/stdout). Please go ahead if you want to work on that.

**Comment by sergiud** [comment]
Thank you @latk and @Spacetown for looking into the issue and for such a great piece of software!

I've attached the verbose log (this time on my local system with build directory set to `build-coverage/`). I failed to mention that my `gcovr.cfg` looks as follows:
```cfg
filter = include/
filter = src/
gcov-parallel = yes
html-details = yes
html-title = Tracker
```
Interestingly, some runs produce the `KeyError: 'Source'` I reported in #588. I've attached both logs.
[gcovr-keyerror-source.log](https://github.com/gcovr/gcovr/files/8365808/gcovr-keyerror-source.log)
[gcovr-file-not-found.log](https://github.com/gcovr/gcovr/files/8365810/gcovr-file-not-found.log)
